From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Thu, 19 Jun 2014 11:37:21 +0200
Subject: [098/249] staging: rtl8723au: rtw_cfg80211_inform_bss(): Use the
 capability info we already saved
Origin: https://git.kernel.org/linus/11a80e8877bf2db8a13b68018049b48e367bd76e

No point in pulling capability info out of the IE array when it's
already stored in struct wlan_bssid_ex

Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/rtl8723au/os_dep/ioctl_cfg80211.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/drivers/staging/rtl8723au/os_dep/ioctl_cfg80211.c b/drivers/staging/rtl8723au/os_dep/ioctl_cfg80211.c
index ff838fc..4eef408 100644
--- a/drivers/staging/rtl8723au/os_dep/ioctl_cfg80211.c
+++ b/drivers/staging/rtl8723au/os_dep/ioctl_cfg80211.c
@@ -247,7 +247,6 @@ static int rtw_cfg80211_inform_bss(struct rtw_adapter *padapter,
 	struct cfg80211_bss *bss;
 	u16 channel;
 	u32 freq;
-	u16 notify_capability;
 	u8 *notify_ie;
 	size_t notify_ielen;
 	s32 notify_signal;
@@ -265,10 +264,6 @@ static int rtw_cfg80211_inform_bss(struct rtw_adapter *padapter,
 
 	notify_channel = ieee80211_get_channel(wiphy, freq);
 
-	notify_capability =
-		get_unaligned_le16(
-			rtw_get_capability23a_from_ie(pnetwork->network.IEs));
-
 	notify_ie = pnetwork->network.IEs + _FIXED_IE_LENGTH_;
 	notify_ielen = pnetwork->network.IELength - _FIXED_IE_LENGTH_;
 
@@ -286,7 +281,7 @@ static int rtw_cfg80211_inform_bss(struct rtw_adapter *padapter,
 	bss = cfg80211_inform_bss(wiphy, notify_channel,
 				  pnetwork->network.MacAddress,
 				  pnetwork->network.tsf,
-				  notify_capability,
+				  pnetwork->network.capability,
 				  pnetwork->network.BeaconPeriod,
 				  notify_ie, notify_ielen,
 				  notify_signal, GFP_ATOMIC);
